apps/dashboard/lib/trpc/routers/key/create.ts

```diff
@@ -120,3 +125,46 @@ export const createKey = t.procedure
 
     return { keyId, key };
   });
+
+async function upsertIdentity(
+  db: Database,
+  workspaceId: string,
+  externalId: string,
+): Promise<Identity> {
+  let identity = await db.query.identities.findFirst({
+    where: (table, { and, eq }) =>
+      and(eq(table.workspaceId, workspaceId), eq(table.externalId, externalId)),
+  });
+  if (identity) {
+    return identity;
+  }
+
+  await db
+    .insert(schema.identities)
+    .values({
+      id: newId("identity"),
+      createdAt: Date.now(),
+      environment: "default",
+      meta: {},
+      externalId,
+      updatedAt: null,
+      workspaceId,
+    })
+    .onDuplicateKeyUpdate({
+      set: {
+        updatedAt: Date.now(),
+      },
+    });
+
+  identity = await db.query.identities.findFirst({
+    where: (table, { and, eq }) =>
+      and(eq(table.workspaceId, workspaceId), eq(table.externalId, externalId)),
+  });
```

Comment:
You should use a catch here incase the db query fails which is different then getting no result.